scripts/install.sh

```diff
@@ -0,0 +1,156 @@
+
+set -e
+
+usage() {
+    echo "
+This script simplifies and automates the installation process of Helm charts on the OpenShift Container Platform (OCP) clusters.
+It ensures that the user is logged into a cluster and attempts to detect the cluster router base, updating the Helm chart configuration accordingly.
+
+Usage:
+  $0 [OPTIONS]
+
+Options:
+  --router-base <router-base> : Manually provide the cluster router base if auto-detection fails.
+  --release-name <name>       : Specify a custom release name for the Helm chart.
+  --generate-name             : Generate a name for the Helm release (overrides --release-name).
+  --namespace <namespace>     : Specify the namespace for the Helm release (autodetects if not provided).
+  --values <file>             : Specify your own values file for the Helm chart.
+  --help                      : Show this help message and exit.
+
+Examples:
+  $0                               # Auto-detects router base and installs the Helm chart
+  $0 --router-base example.com     # Manually specifies the router base and installs the Helm chart
+  $0 --release-name myrelease      # Installs the Helm chart with the specified release name
+  $0 --generate-name               # Generates a name for the Helm release
+  $0 --values /path/to/values.yaml # Installs the Helm chart using the specified values file
+"
+}
+
+# Function to check if a command exists
+command_exists() {
+    command -v "$1" >/dev/null 2>&1
+}
+
+# Check for required commands
+if ! command_exists helm || ! command_exists oc; then
+    echo "Error: Required commands 'helm' and 'oc' are not available."
+    exit 1
+fi
+
+# Check if required files and directories exist
+HELM_CHART_DIR="$(dirname "$0")/../charts/backstage"
+DEFAULT_VALUES_FILE="$HELM_CHART_DIR/values.yaml"
+
+if [ ! -d "$HELM_CHART_DIR" ]; then
+    echo "Error: Helm chart directory not found at $HELM_CHART_DIR"
+    exit 1
+fi
+
+# Parse command-line arguments
+ROUTER_BASE=""
+RELEASE_NAME=""
+GENERATE_NAME=false
+NAMESPACE=""
+VALUES_FILE="$DEFAULT_VALUES_FILE"
+EXTRA_HELM_ARGS=""
+
+while [[ "$#" -gt 0 ]]; do
+    case $1 in
+        --router-base)
+            ROUTER_BASE="$2"
+            shift
+            ;;
+        --release-name)
+            RELEASE_NAME="$2"
+            shift
+            ;;
+        --generate-name)
+            GENERATE_NAME=true
+            ;;
+        --namespace)
+            NAMESPACE="$2"
+            shift
+            ;;
+        --values)
+            VALUES_FILE="$2"
+            shift
+            ;;
+        --help)
+            usage
+            exit 0
+            ;;
+        *)
+            EXTRA_HELM_ARGS+=" $1"
+            ;;
+    esac
+    shift
+done
+
+# Function to detect cluster router base
+detect_cluster_router_base() {
+    ROUTER_BASE=$(oc get ingress.config.openshift.io/cluster -o=jsonpath='{.spec.domain}')
+}
+
+# Detect cluster router base if not provided
+if [[ -z "$ROUTER_BASE" ]]; then
```

Comment:
Just tried this on [Dev Sandbox](https://developers.redhat.com/developer-sandbox), and I was expecting to see the `Error: Cluster router base could not be detected. Please provide it using the --router-base flag.` message, but got a generic `oc` error instead:

```
$  ./scripts/install.sh --generate-name
Error from server (Forbidden): ingresses.config.openshift.io "cluster" is forbidden: User "rh-ee-asoro" cannot get resource "ingresses" in API group "config.openshift.io" at the cluster scope
```

Even when I passed the `--router-base` flag, I got an error:

```
$ ./scripts/install.sh --generate-name --router-base apps.sandbox-m4.g2pi.p1.openshiftapps.com
Error: Cluster router base could not be detected. Please provide it using the --router-base flag.
```

Am I missing something?